asserts/device_asserts.go

```diff
@@ -78,12 +78,18 @@ func (mod *Model) Gadget() string {
 
 // Kernel returns the kernel snap the model uses.
 func (mod *Model) Kernel() string {
-	return mod.HeaderString("kernel")
+	kernel := mod.HeaderString("kernel")
+	return strings.SplitN(kernel, "=", 2)[0]
```

Comment:
if we make =track syntax general in model, it seems longer term  this kind of accessor on model will need to return a struct  with snap-name, track instead of having two accessors. not a blocker for now though